core/deployment/src/main/java/io/quarkus/deployment/pkg/builditem/AppCDSRequestedBuildItem.java

```diff
@@ -13,13 +13,23 @@ public final class AppCDSRequestedBuildItem extends SimpleBuildItem {
     /**
      * Directory where various files needed for AppCDS generation will reside
      */
-    private final Path appCDSDir;
+    private final Path dir;
+    private final JvmStartupOptimizerArchiveType type;
 
-    public AppCDSRequestedBuildItem(Path appCDSDir) {
-        this.appCDSDir = appCDSDir;
+    public AppCDSRequestedBuildItem(Path dir) {
+        this(dir, JvmStartupOptimizerArchiveType.AppCDS);
+    }
+
+    public AppCDSRequestedBuildItem(Path dir, JvmStartupOptimizerArchiveType type) {
+        this.dir = dir;
+        this.type = type;
     }
 
     public Path getAppCDSDir() {
```

Comment:
Ideally, we would change this name but let's do it in a follow-up so that we can make sure we push this one to 3.19.